src/qmozview_p.cpp

```diff
@@ -1472,12 +1478,40 @@ bool QMozViewPrivate::handleAsyncMessage(const QString &message, const QVariant
             }
         }
 
+        return true;
+    } else if (message == QLatin1String(FORMASSIST_RESULT)) {
+        mAutoCompleteActive = true;
+        mAutoCompleteList = data.toStringList();
+        applyAutoCorrect();
+        return true;
+    } else if (message == QLatin1String(FORMASSIST_HIDE)) {
+        mAutoCompleteActive = false;
+        mAutoCompleteList.clear();
+        applyAutoCorrect();
         return true;
     }
 
     return false;
 }
 
+void QMozViewPrivate::applyAutoCorrect()
+{
+    QInputMethod *inputContext = qGuiApp->inputMethod();
+    QVariantMap extensions = q->property("__inputMethodExtensions").toMap();
```

Comment:
I think `"__inputMethodExtensions"` can be used as constant `QStringLiteral`.

Reply:
Thanks for pointing this out as I'm always forgetting. In this case it turns out the `QObject::property()` method actually needs a `const char *name` parameter, so it won't accept a `QStringLiteral` here.